salt/states/boto_iam.py

```diff
@@ -472,16 +476,32 @@ def group_present(name, policies=None, policies_from_pillars=None, users=None, r
     if not _ret['result']:
         ret['result'] = _ret['result']
         return ret
-    # TODO: make this work like _group_policies_present. This is currently not
-    #       fully idempotent.
     if users:
         log.debug('Users are : {0}.'.format(users))
         for user in users:
             log.debug('User is : {0}.'.format(user))
             ret = _case_group(ret, user, name, region, key, keyid, profile)
+        for user in _case_delete(users, exists):
```

Comment:
I dunno about changing user_present. Sometimes you just need a user to be part of a group and don't care who else belongs to that group. It wouldn't make sense to create a user that is the only member of a group. Why create a group in the first place ? 

It does make sense to do it like that in group present though since you have the option to add multiple users. 

But this is just my opinion. If you think this is the correct approach and design I'll make the changes you asked to _case_group